step/serialization.md

```diff
@@ -89,9 +97,9 @@ public class ClassSerializationServer {
 				ByteArrayInputStream byteInput = new ByteArrayInputStream(buffer.flip().array());
```

Comment:
unless it's done somewhere, can you add a high-level overview of what is done here?